intro_calculating_cred_grain.md

```diff
@@ -0,0 +1,162 @@
+
+Additionally, both edges and some nodes have timestamps. Edges are associated with actions or events, and as such, always have a timestamp. For nodes that correspond to a contribution, those nodes have a timestamp corresponding to the moment of the contribution. However, nodes corresponding to contributors do not have time stamps, since the contributors exist in the algorithm across time.
+
+#### Summary up to this point:
+
+Now that we know about components of the weights graph, their properties, and how they fit together, let’s cover how they’re generated.
+
+## Generating the weights graph
+
+#### Plugins:
+
+SourceCred uses APIs of collaboration platforms to gather information about contributions. Currently there are plugins for GitHub and Discourse ([code](https://github.com/sourcecred/sourcecred/tree/master/src/plugins)), and we’re enthusiastic to have more plugins, so we can account for contributions in different venues. This includes Discord (in progress), and could extend to Twitter or citations on the [arXiv](https://arxiv.org/) or [eprint](https://eprint.iacr.org/).
+
+#### Weights:
+
+Because these plugins incorporate data from all sorts of contributions and output a single weighted, directed graph, there needs to be a way to quantitatively compare actions (e.g. authoring a GitHub issue vs responding or referencing it). To accomplish this, there’s a set of constants that determine weights for nodes and edges. For an example set of constants, see the below figure, captured from [sourcecred.io/cred/timeline/@sourcecred/](https://sourcecred.io/cred/timeline/@sourcecred/) (follow the link and click the “Show weight configuration” button).
+
+![](https://lh6.googleusercontent.com/BDSrcvjcCXXfKQ0NSiHIYuAh8fvTaE0Akl0g2-PLdB88fv5TXZS7btltL0YTe77TLOT7nyImct7sPrKATbAQKUwukM3bdxgVfIJCf-xDZo0nvGnFINMoqXq41glKAI9hA2EnThed)
+
+> Figure 3. A set of weight sliders to set parameters controlling the relative weights of different contributions.
+
+Additionally, parameters, cred, and grain are all scoped to a community. This means that the community that supports a project sets and controls the parameters. For that matter, it’s worth reiterating that (lowercase) cred and grain generated for a project are specific to that project. SourceCred is about providing community control, keeping humans in the loop, both for gaming mitigation and general adaptability. At the bottom line, it’s important to keep the humans in control, because good governance technology only works with the consent of the governed.
+
+#### Manual input:
+
+It’s also worth noting that not all contributions occur on platforms with automated data fetching, and thus the capability exists to individually add them to the weighted graph directly at this stage. It’s clear that there will always be data that has to be incorporated manually, and how to incorporate those contributions, especially those not easily quantified, remains an active area of discussion.
+
+#### Merging graphs:
+
```

Comment:
"weights are added together" could be misleading since it implies that we mathematically add them. Really we take the union of the weights, and throw an error on non-disjoint weights.

Reply:
Gotcha.  I fixed this paragraph, and added some motivation on why it should be intuitive that the sets will be disjoint.